wg-batch/README.md

```diff
@@ -0,0 +1,38 @@
+<!---
+This is an autogenerated file!
+
+Please do not edit this file directly, but instead make changes to the
+sigs.yaml file in the project root.
+
+To understand how this file is generated, see https://git.k8s.io/community/generator/README.md
+--->
+# Batch Working Group
+
+Discuss and enhance the support for Batch (eg. HPC, AI/ML, data analytics) workloads in core Kubernetes. We want to unify the way users deploy batch workloads to improve portability and to simplify supportability for Kubernetes providers.
```

Comment:
```suggestion
Discuss and enhance the support for Batch (eg. HPC, AI/ML, CI, data analytics) workloads in core Kubernetes. We want to unify the way users deploy batch workloads to improve portability and to simplify supportability for Kubernetes providers.
```

we should probably explicitly mention CI workloads here too? - Mostly because they have some similar-but-slightly-different traits that would be great to improve.

Reply:
Just to be sure we are talking about the same thing: CI == Continuous Integration?

If so, I'm actually tempted to put it under Non-Goals. Could you mention the challenges these kind of workloads have? Do you expect the solutions to cross SIG boundaries?

Reply:
We are - I'm not sure why that would be a non goal however.

In many cases, they're similar to other batch workloads (1..n pods to represent some work to be done), that may require some coordination when it comes to scheduling constraints. They are however likely to be high frequency but narrow jobs, which can introduce "fun" performance issues.

Including them at least in theory while designing things from the beginning could be helpful for a lot of folks.

Reply:
> We are - I'm not sure why that would be a non goal however.

It feels closer to the existing non-goal: Uses of the batch APIs as support for serving workloads (eg. backups,  upgrades, migrations).
We need to keep a reasonable set of goals to avoid trying to spread us too thin. I know there are a few frameworks that focus on CI, like Tekton (or even Argo). If you could point to some specific problems that you would like to solve in k8s core?

> In many cases, they're similar to other batch workloads (1..n pods to represent some work to be done), that may require some coordination when it comes to scheduling constraints. 

by coordination do you mean "job B depends on job A to have finished"?

> They are however likely to be high frequency but narrow jobs, which can introduce "fun" performance issues.

I think the rest of the types of listed jobs are rather long running, low frequency. So that's a potential difference with CI.
You could still benefit with the performance improvements we could come up with. Or you could work directly with SIG Apps (instead of the Batch WG) to solve this problem.

In summary, I would like to understand what you are asking before we explicitly include this type of workloads. CI is a big world.

Reply:
Like for AI/ML jobs, this is not about job orchestration but actual CI jobs.

Mentioning CI jobs in those goals could make sense since many organizations are transitioning from running CI jobs on bare-metal to Kubernetes and care about resources utilization. Many CI job challenges are common with other types of jobs and for example the work done in https://github.com/kubernetes-sigs/scheduler-plugins helps for a lot of them. That's why it would be nice if the WG can acknowledge that CI jobs are also considered.

One challenge probably more specific to CI jobs than other type of jobs that I'm not sure if it should be in scope for this WG initially is distributed caching (e.g. build caches). We talked about this at the KubeCon last year: https://www.youtube.com/watch?v=wTENRhYt3mw.

Reply:
> One challenge probably more specific to CI jobs than other type of jobs that I'm not sure if it should be in scope for this WG initially is distributed caching (e.g. build caches).

I see. Thanks for mentioning. As I said earlier, CI is a big world. If you can be more specific about which kind of features you are lacking to improve CI jobs, we can came up with a reasonable scope (and a set of non-goals related to CI). Again, I want to avoid spreading us too thin.

Reply:
@alculquicondor Basically similar features that are already in scope like:
- _A performant job controller that can scale to thousands of pods per minute._
- _A set of APIs to support job queueing, a framework to support different queing policies and a ready-to-use implementation as a subproject._
- _Scheduling plugin(s) to support different batch needs._

One of my point was to acknowledge CI jobs are also considered as K8s Batch Workloads. Also, `AI/ML` is a huge world as well, so maybe if we mention it then we can mention CI as well?

Reply:
That brings me that I'm missing some points related to AI/ML that we should explicitly exclude :)

How would you categorize a feature such as "build cache"? I think it should be out of scope, but I want to make the point more general.

Reply:
I added it for now. Let me know if you can think of a type of features we can exclude.

Reply:
Should we add ETL to the list?   Many users are moving this type of workload from the mainframe or existing on-metal systems to Kubernetes deployments.

Reply:
I think it's already covered as part of "data analytics"

Reply:
@alculquicondor I guess, but that is a stretch.   Many ETL apps are not for analytics but rather data processing.

Reply:
is there a term that includes both? "data processing" to me sounds like it should include "analytics" and ETL. I'm sure more people will come and ask to include their "type" of batch job. But to my eyes, this is already covered. We just have to find a better wording.

Reply:
Agreed.   ETL is too tight.   I was trying to find an acronym for Data Processing or Information Processing, for those folks writing these types of apps.

Reply:
+1 (sorry I'm still on PTO, so am kinda back and forth)